pkg/operator/apiservicecontroller/apiservice_controller.go

```diff
@@ -230,3 +248,107 @@ func (c *APIServiceController) eventHandler() cache.ResourceEventHandler {
+		}
+
+		apiServicesToManage := []*apiregistrationv1.APIService{}
+		for _, apiService := range authoritativeAPIServicesList {
+			if apiGroupsManagedByExternalServer.Has(apiService.Name) && isAPIServiceAnnotatedByExternalServer(apiregistrationv1Client, apiService, apiGroupsManagedByExternalServerAnnotation) {
+				continue
+			}
+			apiServicesToManage = append(apiServicesToManage, apiService)
+		}
+
+		{
+			added, removed := diffAPIServices(observedAPIServices, apiServicesToManage)
+			if len(added) > 0 {
+				eventRecorder.Eventf("OAuthAPIResourcesManagement", "The API Services for the following GVs %v will be managed by this operator (CAS-O)", added)
+				klog.V(2).Infof("The API Services for the following GVs %v will be managed by this operator (CAS-O)", added)
+			}
+
+			if len(removed) > 0 {
+				eventRecorder.Eventf("OAuthAPIResourcesManagement", "The API Services for the following GVs %v will be managed by an external operator/server", removed)
+				klog.V(2).Infof("The API Services for the following GVs %v will be managed by an external operator/server", removed)
+			}
+		}
+
+		observedAPIServices = apiServicesToManage
+		return observedAPIServices, nil
+	}
+}
+
+func isAPIServiceAnnotatedByExternalServer(apiregistrationv1Client apiregistrationv1client.ApiregistrationV1Interface, apiService *apiregistrationv1.APIService, apiGroupsManagedByExternalServerAnnotation string) bool {
+	existingApiService, err := apiregistrationv1Client.APIServices().Get(apiService.Name, metav1.GetOptions{})
+	if err != nil {
+		klog.V(4).Infof("unable to determine if the following API Service %s was annotated by an external operator due to %v", apiService.Name, err)
+		return false
+	}
+
+	if _, ok := existingApiService.Annotations[apiGroupsManagedByExternalServerAnnotation]; ok {
+		return true
+
+	}
+	return false
+}
+
+func diffAPIServices(old []*apiregistrationv1.APIService, new []*apiregistrationv1.APIService) (added []string, removed []string) {
+	oldSet := sets.String{}
+	for _, oldService := range old {
+		oldSet.Insert(oldService.Name)
+	}
+
+	newSet := sets.String{}
+	for _, newService := range new {
+		newSet.Insert(newService.Name)
+	}
+
+	removed = oldSet.Difference(newSet).List()
+	added = newSet.Difference(oldSet).List()
+
+	return added, removed
+}
+
+// AuthenticationOperatorAvailableAndConditionSetPrecondition lets the authentication operator know that he will manage OAuth API Resources by setting the appropriate condition on its status
+func AuthenticationOperatorAvailableAndConditionSetPrecondition(authOperatorName string, authOperatorGVR schema.GroupVersionResource, authOperatorConditionType string, authOperatorClient v1helpers.OperatorClient) func() error {
+	return func() error {
+		_, operatorStatus, _, err := authOperatorClient.GetOperatorState()
+		if err != nil {
+			return err
+		}
+
+		if !v1helpers.IsOperatorConditionTrue(operatorStatus.Conditions, operatorv1.OperatorStatusTypeAvailable) {
```

Comment:
I don't think available enters into it, but this is the spot to put `ManagingOAuthAPIServer`

Reply:
this is what you had in your KEP :)